wsgi_kerberos.py

```diff
@@ -49,37 +56,47 @@ def _consume_request(environ, read_max):
     '''
     if read_max == 0:  # Short-circuit early when user opts out of this.
         return
+    if environ["REQUEST_METHOD"] in _NEVER_READ_METHODS:
+        return
     if environ.get("HTTP_EXPECT") == "100-continue":
         return
     try:
-        sock = environ.get('wsgi.input')
-        if hasattr(sock, 'closed') and sock.closed:
+        body = environ.get('wsgi.input')
+        if hasattr(body, 'closed') and body.closed:
             return
 
         content_length = environ.get('CONTENT_LENGTH', '')
         if not content_length:
-            remaining = read_max
-        else:
-            content_length = int(content_length)
-            if content_length > read_max:
-                # User is not willing to read such a large request body, but
-                # reading anything less does not help naively-written clients.
-                return
-            remaining = content_length
-
+            LOG.debug("No Content-Length -> skipping _consume_request")
+            return
+        content_length = int(content_length)
+        if content_length > read_max:
+            # Server is not willing to read such a large request body, but
+            # reading anything less does not help naively-written clients.
+            LOG.warning(
+                "Content-Length (%s) exceeds read_max (%s) -> not consuming"
+                " request body; client may get a connection error. Enabling"
+                " preemptive authentication on the client may avoid this."
+                " You can also pass a higher value of `read_max_on_auth_fail`"
+                " to KerberosAuthMiddleware.",
+                content_length, read_max
+            )
+            return
         # Try to receive all of the data. Keep retrying until we get an error
         # which indicates that we can't retry. Eat errors. The client will just
         # have to deal with a possible Broken Pipe -- we tried.
+        remaining = content_length
         while remaining > 0:
+            to_read = min(io.DEFAULT_BUFFER_SIZE, remaining)
             try:
-                delta = len(sock.read(remaining))
+                read = len(body.read(to_read))
             except socket.error as err:
                 if err.errno != errno.EAGAIN:
-                    break
+                    raise
```

Comment:
I think this should be `raise` here so we at least log the (non-EAGAIN) socket error at debug level below, rather than swallow it completely. Still go back to the top of the `while` loop to try again in the EAGAIN case.

Reply:
Sounds good